.changes/next-release/bugfix-NettyNIOAsyncHTTPClient-5b20e7b.json

```diff
@@ -0,0 +1,5 @@
+{
+    "category": "Netty NIO Async HTTP Client", 
+    "type": "bugfix", 
+    "description": "Fix the issue where PutObjectRequest sometimes is hanging because 100Continue response message is not being read automatically. See [#459]"
```

Comment:
nit: can we make it more general instead of PutObjectRequest in case it also affects other similar operations?

Reply:
It's only affecting `PutObjectRequest` because we only add `Expect: 100-continue` header to `PutObjectRequest`. 

https://github.com/aws/aws-sdk-java-v2/blob/d314a0d15fb8f80e3653beca3f8c216bf8767d8d/services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/handlers/PutObjectInterceptor.java#L35

But yeah I can make it more general